docs/src/main/sphinx/develop/connectors.rst

```diff
@@ -393,13 +403,109 @@ and our ``ConstraintApplicationResult`` can set ``remainingFilter`` to
+        if (newDomain.isNone()) {
+            newConstraintExpressions = ImmutableList.of();
+            remainingFilter = TupleDomain.all();
+            remainingExpression = Optional.of(Constant.TRUE);
+        }
+        else {
+            // We need to decide which columns to push down.
+            // Since this is a base class for many JDBC-based connectors, each
+            // having different Trino type mappings and comparison semantics
+            // it needs to be flexible.
+
+            Map<ColumnHandle, Domain> domains = newDomain.getDomains().orElseThrow();
+            List<JdbcColumnHandle> columnHandles = domains.keySet().stream()
+                    .map(JdbcColumnHandle.class::cast)
+                    .collect(toImmutableList());
+
+            // Get information about how to push down every column based on it's
+            // JDBC data type
+            List<ColumnMapping> columnMappings = jdbcClient.toColumnMappings(
+                    session,
+                    columnHandles.stream()
+                            .map(JdbcColumnHandle::getJdbcTypeHandle)
+                            .collect(toImmutableList()));
+
+            // Calculate the domains which can be safely pushed down (supported)
+            // and those which need to be filtered in Trino (unsupported)
+            Map<ColumnHandle, Domain> supported = new HashMap<>();
+            Map<ColumnHandle, Domain> unsupported = new HashMap<>();
+            for (int i = 0; i < columnHandles.size(); i++) {
+                JdbcColumnHandle column = columnHandles.get(i);
+                DomainPushdownResult pushdownResult =
+                    columnMappings.get(i).getPredicatePushdownController().apply(
+                        session,
+                        domains.get(column));
+                supported.put(column, pushdownResult.getPushedDown());
+                unsupported.put(column, pushdownResult.getRemainingFilter());
+            }
+
+            newDomain = TupleDomain.withColumnDomains(supported);
+            remainingFilter = TupleDomain.withColumnDomains(unsupported);
+        }
+
+        // Return empty Optional if nothing changed in filtering
+        if (oldDomain.equals(newDomain)) {
+            return Optional.empty();
+        }
+
+        handle = new JdbcTableHandle(
+                handle.getRelationHandle(),
+                newDomain,
+                ...);
+
+        return Optional.of(
+                new ConstraintApplicationResult<>(
+                    handle,
+                    remainingFilter));
+    }
+
+In this example we are implementing a base class for many JDBC connectors,
+so we have to handle all the nuances of different JDBC-capable databases.
+We need to ensure that if a constraint gets pushed down, it should work
+exactly the same in the underlying data source and produce the same results
+as it would in Trino.
+
+For example, string comparison in some databases is case-insensitive, which
```

Comment:
Move this to previous paragraph. 

Do not use "we" - too informal.

For example, in databases where string comparisons are is case-insensitive, pushdown does not work, as string
comparison operations in Trino are case-sensitive.